src/devices/generic/device.ts

```diff
@@ -1,10 +1,17 @@
 import { EventEmitter } from "events";
 
-import { IDeviceInterface, IDeviceMode } from "../interfaces";
+import { IDeviceInterface, IDeviceMode } from "../../interfaces";
 
-import * as Consts from "../consts";
+import { BLECharacteristic, HubType, ValueBits, ValueType } from "../../consts";
+
+export type DeviceVersion = {
+    hardware: string,
+    software: string,
+}
```

Comment:
Add hardware/software version to handle modes from "ability" configuration files